src/librustdoc/lib.rs

```diff
@@ -16,6 +16,12 @@
 
 #[macro_use]
 extern crate lazy_static;
+#[macro_use]
+extern crate tracing;
+
+// N.B. these need `extern crate` even in 2018 edition
+// because they're loaded implicitly from the sysroot.
+// Dependencies listed in Cargo.toml do not need extern crate.
```

Comment:
So the question now is why does rustdoc have to search dependencies in sysroot rather than specify them in `Cargo.toml`.

Reply:
I think it's because we don't build it in the rustc cargo directory, in part because that gives us finer-grained deleting. That means if we specified things in Cargo.toml, we'd end up rebuilding rustc crates.

Reply:
I updated the comment to mention why rustdoc uses the sysroot instead of cargo.toml.